api4/user.go

```diff
@@ -1493,16 +1493,23 @@ func updatePassword(c *Context, w http.ResponseWriter, r *http.Request) {
 	defer c.LogAuditRec(auditRec)
 	c.LogAudit("attempted")
 
+	var canUpdatePassword bool
 	if user, err := c.App.GetUser(c.Params.UserId); err == nil {
 		auditRec.AddMeta("user", user)
+
+		if user.IsSystemAdmin() {
+			canUpdatePassword = c.App.SessionHasPermissionTo(*c.App.Session(), model.PERMISSION_MANAGE_SYSTEM)
```

Comment:
If a system admin, do we still need check if have manage system permission or can we just set it to ture?

Reply:
This is checking if the user who's `password` is being changed is a `system_admin`, not checking that the user performing the request is a sys admin... so basically if you want to change the password of a sysadmin you have to be a sysadmin yourself